### PR TITLE
Separate form submission processing phases

### LINKED ISSUE #1358: Separate normalized form actions from conversion tracking and frontend effects
## Outcome

Let extensions action a normalized form success while independently suppressing or allowing Popup Maker conversion tracking and frontend effects.

Cross-repository parent: https://github.com/PopupMaker/Pro/issues/132

## Problem

The current processing guard conflates AJAX/REST transport constraints with whether a submission may be captured, actioned, tracked as a conversion, persisted, or allowed to trigger frontend behavior. Pro needs to run server-side actions without fabricating conversions or requiring browser execution.

## Scope

- Define independent decisions/phases for:
  - normalize/capture;
  - dispatch extension actions;
  - conversion tracking;
  - optional storage;
  - frontend effects.
- Preserve existing behavior by default.
- Add narrowly named filters/actions rather than embedding Pro concepts.
- Make repeated delivery of the same normalized event idempotent for Core-owned tracking.
- Document server-only, AJAX, REST, redirect, and browser-success behavior.

## Out of scope

- Pro action implementations.
- Submission tables or profile storage.
- Provider field schemas.
- Lead Magnet claims or delivery.
- Changing analytics definitions.

## Acceptance criteria

- An extension can process a valid form submission while explicitly bypassing conversion tracking.
- Conversion tracking can run without requiring storage.
- A browser event cannot cause the same Core conversion to count twice.
- Existing Analytics and Split Testing attribution remains unchanged unless an extension opts into a new seam.
- Tests cover server callbacks, frontend callbacks, duplicate events, and opt-out combinations.

## Expected PR

One Core PR after the normalized context contract.

### PR DESCRIPTION
**Pull request by danieliser**
## Summary

- add a product-neutral `phases` policy with independent `actions`, `tracking`, and `frontend` authorization to normalized PHP and JavaScript form submissions
- let AJAX and REST provider callbacks reach normalized observation and server action hooks while keeping Core conversion tracking and frontend replay disabled by default
- preserve unconditional existing observation hooks and add gated runtime-specific action hooks for extensions
- make non-AJAX browser replay a tracking/action receipt while retaining cookies, form triggers, and popup close behavior
- suppress repeated native browser submission keys with a bounded in-memory cache without claiming cross-transport correlation for generated UUIDs
- preserve provider-native entry IDs from Fluent Forms, Gravity Forms, Formidable Forms, WPForms, and Bit Form where their callbacks expose one

## Why

The legacy `should_process_submission()` guard rejected every AJAX or REST callback before normalization. That coupled transport to unrelated decisions: an extension could not action a provider-confirmed success unless Core also treated it like a browser conversion, and the same coarse guard controlled whether browser behavior could replay.

This split lets Pro and other extensions observe or action server successes without fabricating Popup Maker conversions. Existing Analytics and Split Testing attribution remain on the existing conversion path unless tracking is explicitly authorized.

## Contract

The normalized envelope now contains:

```php
'phases' => [
	'actions'  => true,
	'tracking' => true,
	'frontend' => true,
]
```

New extension points:

- PHP filter: `pum_integrated_form_submission_phases`
- PHP action: `pum_integrated_form_submission_actions`
- JavaScript filter: `pum.integration.form.phases`
- JavaScript action: `pum.integration.form.actions`

The existing `pum_integrated_form_submission` and `pum.integration.form.success` hooks remain unconditional observation hooks. A `tracked: true` receipt cannot opt back into tracking.

PHP and JavaScript action hooks are deliberately runtime-specific. AJAX providers do not consistently expose a native server receipt to their browser event, so consumers must choose the runtime appropriate to their action or provide their own cross-runtime idempotency when intentionally listening to both. Generated IDs remain event-local.

## Default behavior

| Path | Actions | Tracking | Frontend |
| --- | --- | --- | --- |
| Normal PHP callback | on | on | on |
| AJAX/REST PHP callback | on | off | off |
| Direct browser success | on | on | on |
| Localized non-AJAX replay | off | off | on |
| Repeated native browser key | off | off | off |

Core form analytics now checks tracking authorization before queueing its beacon. Core form cookies, popup close behavior, and form-triggered popups check frontend authorization. The legacy popup conversion counter and newer form-conversion counters continue to increment exactly once on their authorized path.

## Validation

- Jest: 2 focused suites, 10 tests pass
- PHPUnit against isolated `pum_core_1358_tests`: 37 focused tests, 115 assertions pass
- PHPCS: no errors in changed PHP files; two existing `error_log()` warnings remain in `FormConversionTracking`
- ESLint: normalized integration code and both focused Jest suites pass
- Markdown lint: developer documentation passes
- PHP syntax and `git diff --check`: pass

Closes #1358

Stacked on #1359. Cross-repository parent: PopupMaker/Pro#132.

### CODE REVIEW

**Comment by danieliser** [comment]
@codex review

File: classes/Integration/Form/GravityForms.php [hunk 1]
```diff
@@ -73,12 +73,6 @@ public function on_success( $entry, $form ) {
 			return;
 		}
 
-		// This key is set when Gravity Forms is submitted via AJAX. Ignored because this is a simple boolean check.
-		// phpcs:ignore WordPress.Security.NonceVerification.Missing
-		if ( isset( $_POST['gform_ajax'] ) && ! is_null( $_POST['gform_ajax'] ) ) {
-			return;
-		}
-
 		$popup_id = $this->get_popup_id();
```
[the remark this answers is not in the document]
Comment (danieliser) [reply classes/Integration/Form/GravityForms.php:83]: Fixed in 07578a312. Gravity Forms now passes its provider-level `gform_ajax` transport marker into the normalized envelope, and the shared phase resolver treats `ajax: true` as asynchronous even outside `admin-ajax.php`. The regression test verifies actions remain available while server tracking and frontend replay stay disabled.

File: classes/Abstract/Integration/Form.php [hunk 2]
```diff
@@ -72,21 +72,33 @@ public function get_popup_id() {
 	 * @since 1.13.0
 	 */
 	public function increase_conversion( $popup_id ) {
+		$phases = pum_get_integrated_form_submission_phases(
+			[
+				'popup_id'      => $popup_id,
+				'form_provider' => $this->key,
+			]
```
[the remark this answers is not in the document]
Comment (danieliser) [reply classes/Abstract/Integration/Form.php]: Fixed in 07578a312. The legacy popup conversion increment now happens centrally after the complete submission envelope and phase policy are resolved. Provider-side pre-envelope increments were removed, so the phase filter runs once and the legacy plus form counters consume the same tracking decision. Regression coverage filters on form ID, submission ID, and context and verifies both counters remain unchanged.

**Comment by danieliser** [comment]
@codex review

**Comment by danieliser** [comment]
@codex review

**Comment by danieliser** [comment]
@codex review

**Comment by danieliser** [comment]
@codex review

**Comment by danieliser** [comment]
@codex review

**Comment by danieliser** [comment]
@codex review

**Comment by danieliser** [comment]
@codex review
